test/e2e/serial/workload_placement_test.go

```diff
@@ -0,0 +1,283 @@
+		err := e2efixture.Teardown(fxt)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("cluster with at least a worker node suitable", func() {
+		It("[placement][case:1] should keep the pod pending if not enough resources available, then schedule when resources are freed", func() {
+			// note we hardcode the values we need here and when we pad node.
+			// This is ugly, but automatically computing the values is not straightforward
+			// and will we want to start lean and mean.
+
+			// make sure this is > 1 and LESS than required Res!
+			unsuitableFreeRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2"),
+				corev1.ResourceMemory: resource.MustParse("4Gi"),
+			}
+
+			requiredRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("4"),
+				corev1.ResourceMemory: resource.MustParse("8Gi"),
+			}
+
+			By(fmt.Sprintf("creating test pod, total resources required %s", e2enrt.ResourceListToString(requiredRes)))
+
+			// TODO: we need AT LEAST 2 (so 4, 8 is fine...) but we hardcode the padding logic to keep the test simple,
+			// so we can't support ATM zones > 2. HW with zones > 2 is rare anyway, so not to big of a deal now.
+			By(fmt.Sprintf("filtering available nodes with at least %d NUMA zones", 2))
+			nrtCandidates := e2enrt.FilterZoneCountEqual(nrts, 2)
+			if len(nrtCandidates) < 1 {
+				Skip(fmt.Sprintf("not enough nodes with 2 NUMA Zones: found %d", len(nrtCandidates)))
+			}
+			By("filtering available nodes with allocatable resources on each NUMA zone that can match request")
+			nrtCandidates = e2enrt.FilterAnyZoneMatchingResources(nrtCandidates, requiredRes)
+			if len(nrtCandidates) < 1 {
+				Skip(fmt.Sprintf("not enough nodes with NUMA zones each of them can match requests: found %d", len(nrtCandidates)))
+			}
+
+			candidateNodeNames := e2enrt.AccumulateNames(nrtCandidates)
+			// nodes we have now are all equal for our purposes. Pick one at random
+			// TODO: make sure we can control this randomness using ginkgo seed or any other way
+			targetNodeName, ok := candidateNodeNames.PopAny()
+			Expect(ok).To(BeTrue(), "cannot select a target node among %#v", candidateNodeNames.List())
+			unsuitableNodeNames := candidateNodeNames.List()
+
+			By(fmt.Sprintf("selecting target node %q and unsuitable nodes %#v (random pick)", targetNodeName, unsuitableNodeNames))
+			var targetPaddingPods []*corev1.Pod
+			var paddingPods []*corev1.Pod
+
+			By(fmt.Sprintf("preparing target node %q to fit the test case", targetNodeName))
+			// first, let's make sure that ONLY the required res can fit in either zone on the target node
+			nrtInfo, err := e2enrt.FindFromList(nrtList.Items, targetNodeName)
+			Expect(err).ToNot(HaveOccurred(), "missing NRT info for %q", targetNodeName)
+
+			for _, zone := range nrtInfo.Zones {
+				By(fmt.Sprintf("padding node %q zone %q", nrtInfo.Name, zone.Name))
+				padPod, err := makePaddingPod(fxt.Namespace.Name, "target", zone, requiredRes)
```

Comment:
isn't that a duplicate behavior of the padder package?

Reply:
it is, we will need to reconcile the code and extract functionalities later.